playback/src/audio_backend/pulseaudio.rs

```diff
@@ -88,20 +132,24 @@ impl Sink for PulseAudioSink {
 
 impl SinkAsBytes for PulseAudioSink {
     fn write_bytes(&mut self, data: &[u8]) -> io::Result<()> {
-        if let Some(s) = &self.s {
-            match s.write(data) {
-                Ok(_) => Ok(()),
-                Err(e) => Err(io::Error::new(
-                    io::ErrorKind::BrokenPipe,
-                    e.to_string().unwrap(),
-                )),
+        match &self.s {
+            Some(s) => {
+                if let Err(e) = s.write(data) {
+                    return Err(io::Error::new(
+                        io::ErrorKind::BrokenPipe,
+                        PulseError::OnWrite(e),
+                    ));
+                }
+            }
+            None => {
+                return Err(io::Error::new(
+                    io::ErrorKind::NotConnected,
+                    PulseError::ServerNone,
+                ));
```

Comment:
Same here.

Reply:
Same. I can do that.

Reply:
I'll get on all these changes after work.

Reply:
Again, kinda what you wanted but more short and sweet I think.